Source/Details/_ASDisplayView.mm

```diff
@@ -358,14 +359,79 @@ - (void)tintColorDidChange
   [node tintColorDidChange];
 }
 
-- (BOOL)canBecomeFirstResponder {
-  ASDisplayNode *node = _asyncdisplaykit_node; // Create strong reference to weak ivar.
-  return [node canBecomeFirstResponder];
+#pragma mark UIResponder Handling
+
+#define HANDLE_RESPONDER_METHOD(__sel) \
+  ASDisplayNode *node = _asyncdisplaykit_node; \
+  SEL sel = @selector(__sel); \
+  /* Prevent an infinite loop in here if [super canBecomeFirstResponder] was called on a
+  / _ASDisplayView subclass */ \
+  if (ASSubclassOverridesSelector([_ASDisplayView class], [self class], sel)) { \
```

Comment:
I think these checks are going to become costly; we should verify in a profiler. Since only a handful of selectors are involved, ideally these should be performed at +initialize like other subclass implementation checks are.

Reply:
Moved them into `+initialize` same way as `ASDisplayNode` does.